frontend/src/app/projects/service/project.service.ts

```diff
@@ -119,10 +119,14 @@ export type UserMetadata = {
   subscribers: number;
 };
 
-export type PatchProject = {
-  name?: string;
-  description?: string;
-  visibility?: ProjectVisibility;
+export type PostProject = {
+  name: string;
+  description: string;
+  visibility: ProjectVisibility;
+};
+
+export type PatchProject = Partial<PostProject> & {
```

Comment:
What is the `Partial` used for?

Reply:
When we create a project all three properties (i.e., name, description, and visibility) are required. However, when we want to patch/update a project, we do not always want to provide all of these properties but only those we actually want to update. So `Partial` in this case, simply returns a type where all properties of, in this case `PostProject`, are optional.

Reply:
Thanks for the explanation :)